pkg/application/blueprintSpecChangeUseCase.go

```diff
@@ -99,6 +99,24 @@ func (useCase *BlueprintSpecChangeUseCase) HandleChange(ctx context.Context, blu
 	}
 }
 
+func (useCase *BlueprintSpecChangeUseCase) handleNew(ctx context.Context, blueprintId string, blueprintSpec *domain.BlueprintSpec) error {
+	err := useCase.addPotentialDryRunEvent(ctx, blueprintSpec)
+	if err != nil {
+		return fmt.Errorf("failed adding dry run event: %w", err)
+	}
+
+	return useCase.validateStatically(ctx, blueprintId)
+}
+
+func (useCase *BlueprintSpecChangeUseCase) addPotentialDryRunEvent(ctx context.Context, blueprintSpec *domain.BlueprintSpec) error {
+	if !blueprintSpec.Config.DryRun {
+		return nil
+	}
+	blueprintSpec.Events = append(blueprintSpec.Events, domain.BlueprintDryRunEvent{})
```

Comment:
do not modify fields of the aggregate outside of the aggregate. In DDD aggregates should only be modified by there own commands.Especially events are always results of commands.

Reply:
Maybe you could add the event in the moment when the further evaluation of the blueprint is stopped because of it, so you could add it in the ApplyBlueprintUseCase. This way, you also don't need to mock as much in the tests. This ChangeUseCase-file here can then encapsulate only the stepping logic of the Blueprint state machine.